src/EPR.SubsidiaryBulkUpload.Application/Services/BulkUploadOrchestration.cs

```diff
@@ -65,39 +66,73 @@ public async Task NotifyErrors(IEnumerable<CompaniesHouseCompany> data, UserRequ
 
-        var subsidiaryGroupsAndParentOrgWithValidCompaniesHouseNumber = subsidiaryGroupsAndParentOrg.Where(sg => sg.parentOrg != null && sg.SubsidiaryGroup.Parent.companies_house_number == sg.parentOrg.companiesHouseNumber);
+        var subsidiaryGroupsAndParentOrgWithValidCompaniesHouseNumber = subsidiaryGroupsAndParentOrg.Where(sg => sg.ParentOrg != null && sg.SubsidiaryGroup.Parent.companies_house_number == sg.ParentOrg.companiesHouseNumber);
 
         var addedSubsidiariesCount = 0;
 
         foreach (var subsidiaryGroupAndParentOrg in subsidiaryGroupsAndParentOrgWithValidCompaniesHouseNumber)
         {
-            addedSubsidiariesCount += await childProcessor.Process(
-                subsidiaryGroupAndParentOrg.SubsidiaryGroup.Subsidiaries,
+            var subsidiariesToProcess = await FilterDuplicateSubsidiaries(userRequestModel, subsidiaryGroupAndParentOrg);
+
+            addedSubsidiariesCount += await _childProcessor.Process(
+                subsidiariesToProcess,
                 subsidiaryGroupAndParentOrg.SubsidiaryGroup.Parent,
-                subsidiaryGroupAndParentOrg.parentOrg,
+                subsidiaryGroupAndParentOrg.ParentOrg,
                 userRequestModel);
         }
 
         await _notificationService.SetStatus(userRequestModel.GenerateKey(NotificationStatusKeys.SubsidiaryBulkUploadProgress), "Finished");
         await _notificationService.SetStatus(userRequestModel.GenerateKey(NotificationStatusKeys.SubsidiaryBulkUploadRowsAdded), addedSubsidiariesCount.ToString());
     }
 
+    private async Task<List<CompaniesHouseCompany>> FilterDuplicateSubsidiaries(
+        UserRequestModel userRequestModel,
+        (ParentAndSubsidiaries SubsidiaryGroup, OrganisationResponseModel? ParentOrg) subsidiaryGroupAndParentOrg)
+    {
+        var duplicatesGroup = subsidiaryGroupAndParentOrg.SubsidiaryGroup.Subsidiaries.GroupBy(companiesHouseCompany => new
+        {
+            companiesHouseCompany.subsidiary_id,
+            companiesHouseCompany.organisation_name,
+            companiesHouseCompany.companies_house_number,
+            companiesHouseCompany.parent_child
+        }).Where(grouping => grouping.Key.parent_child.ToLowerInvariant() == "child").ToList();
+
+        var subsidiariesToProcess = subsidiaryGroupAndParentOrg.SubsidiaryGroup.Subsidiaries;
+
+        if (duplicatesGroup.Exists(g => g.Count() > 1))
+        {
+            var duplicateItems = subsidiaryGroupAndParentOrg.SubsidiaryGroup.Subsidiaries.Where(company =>
+
+                company.subsidiary_id == duplicatesGroup[0].Key.subsidiary_id &&
+                company.organisation_name == duplicatesGroup[0].Key.organisation_name &&
+                company.companies_house_number == duplicatesGroup[0].Key.companies_house_number &&
```

Comment:
this block required to be excluding the duplicates from the remaining rows. duplicates shouldn't be sent for process.

Reply:
The RemoveAt will remove the first record (which we want to preserve) and the line 

`subsidiariesToProcess = subsidiariesToProcess.Except(duplicateItems).ToList();`

will only send the first record only, not the duplicates